### DISCUSSION TITLE
Issues with `brew bump-formula-pr`

### DISCUSSION THREAD

**Discussion by krushnarout** [Homebrew/discussions#5753] [Tap maintenance and Homebrew development] [answered]
### Output of `brew config`

```shell
na
```

### Output of `brew doctor`

```shell
na
```

### Description of issue

Hello everyone,  

I’m facing an issue while automating the Homebrew formula update for [Screenpipe Formula](https://github.com/Homebrew/homebrew-core/blob/master/Formula/s/screenpipe.rb). Here's what I’ve done so far:

### Context:
I’ve integrated the `brew bump-formula-pr` method into our GitHub Actions workflow for automatic formula updates. The relevant code is here: 

https://github.com/mediar-ai/screenpipe/blob/main/.github/workflows/release-cli.yml#L102)](https://github.com/mediar-ai/screenpipe/blob/main/.github/workflows/release-cli.yml#L102) 

Here’s a snippet of the workflow:  

```yaml
- name: Update Homebrew Formula
  env:
    GH_TOKEN: ${{ secrets.PAT }}
    HOMEBREW_GITHUB_API_TOKEN: ${{ secrets.PAT }}
  run: |
    # Install Homebrew
    /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
    
    # Setup Homebrew environment
    eval "$(/opt/homebrew/bin/brew shellenv)"
    
    # Prepare URL and SHA
    URL="https://github.com/${GITHUB_REPOSITORY}/releases/download/v${{ env.VERSION }}/screenpipe-${{ env.VERSION }}-${{ matrix.target }}.tar.gz"
    SHA=$(echo "$MAC_SHA256_${{ matrix.target }}")
    
    # Create PR using brew bump-formula-pr
    brew bump-formula-pr screenpipe \
      --version=${{ env.VERSION }} \
      --url="${URL}" \
      --sha256="${SHA}" \
      --message="Automated update to version ${{ env.VERSION }} for ${{ matrix.target }}" \
      --no-browse \
      --no-audit
```

After tagging and pushing a new version (e.g., `git tag v<my version>` and `git push origin v<my version>`), I encountered the following issues:  

1. **Duplicate PRs Error**:

https://github.com/mediar-ai/screenpipe/actions/runs/11958214212/job/33338214634#step:11:69

```
Error: These pull requests are duplicates:
screenpipe 0.2.0 https://github.com/Homebrew/homebrew-core/pull/198523
Duplicate PRs must not be opened.
Manually open these PRs if you are sure that they are not duplicates (and tell us that in the PR).
```

2. **Incorrect SHA256**:

[Screenpipe 0.2.0 #198523](https://github.com/Homebrew/homebrew-core/pull/198523)
   
The PR only updates the URL but doesn’t update the SHA256 correctly.

To address the SHA256 issue, I modified the workflow to calculate the SHA256.

```yaml
# Updated URL and SHA calculation
URL="https://github.com/${GITHUB_REPOSITORY}/archive/refs/tags/v${VERSION}.tar.gz"
curl -L -o source.tar.gz "${URL}"
SHA=$(shasum -a 256 source.tar.gz | awk '{print $1}')

# Create PR using brew bump-formula-pr
brew bump-formula-pr screenpipe \
  --url="${URL}" \
  --sha256="${SHA}" \
  --message="Automated update to version ${VERSION}" \
  --no-browse \
  --no-audit
```

But I’m still unclear if this resolves the correctly counting the `sha256`. 

I’ve found very little documentation on `brew bump-formula-pr`, so any guidance would be much appreciated.  

Thank you!

**Comment by SMillerDev** [comment]
Why not use an existing action like:
- https://github.com/marketplace/actions/homebrew-bump-formula
- https://github.com/marketplace/actions/bump-homebrew-formula

---

The duplicate PR was because someone else also made a PR, there's not much you can do about that.

---

> I’ve found very little documentation on brew bump-formula-pr, so any guidance would be much appreciated.
Was this documentation in the manpages not sufficient?

https://docs.brew.sh/Manpage#bump-formula-pr-options-formula

**Reply by krushnarout** [reply]
> Why not use an existing action like:
> 
> * https://github.com/marketplace/actions/homebrew-bump-formula
> * https://github.com/marketplace/actions/bump-homebrew-formula
> 

Thanks! Do these actions work for Linux?

> The duplicate PR was because someone else also made a PR, there's not much you can do about that.
> 

Now, if they release a new version, would they need to use --force due to [bump-formula-pr: search for closed dupe PRs](https://github.com/Homebrew/brew/pull/7745 )?

> > I’ve found very little documentation on brew bump-formula-pr, so any guidance would be much appreciated.
> > Was this documentation in the manpages not sufficient?
> 
> https://docs.brew.sh/Manpage#bump-formula-pr-options-formula

Yes, the documentation is sufficient. I followed it to write the code, but it would be great if there were some examples.

**Reply by SMillerDev** [reply, marked as answer]
> Thanks! Do these actions work for Linux?

They work anywhere, just like brew does.

> Now, if they release a new version, would they need to use --force due to https://github.com/Homebrew/brew/pull/7745?

No, there should just be one source of automated pull requests. If someone else set this up, you should not.

> it would be great if there were some examples.

There are a bunch of them in: https://github.com/search?q=brew+bump-formula-pr&type=code

**Reply by krushnarout** [reply]
Thank you so much!

**Reply by krushnarout** [reply]
Lastly, is this code correct or not? They think my code is wrong.

```
- name: Update Homebrew Formula
  env:
    GH_TOKEN: ${{ secrets.PAT }}
    HOMEBREW_GITHUB_API_TOKEN: ${{ secrets.PAT }}
  run: |
    # Install Homebrew
    /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
    
    # Setup Homebrew environment
    eval "$(/opt/homebrew/bin/brew shellenv)"
    
    # Updated URL and SHA calculation
     URL="https://github.com/${GITHUB_REPOSITORY}/archive/refs/tags/v${VERSION}.tar.gz"
     curl -L -o source.tar.gz "${URL}"
     SHA=$(shasum -a 256 source.tar.gz | awk '{print $1}')

     # Create PR using brew bump-formula-pr
      brew bump-formula-pr screenpipe \
         --url="${URL}" \
         --sha256="${SHA}" \
         --message="Automated update to version ${VERSION}" \
         --no-browse \
         --no-audit
 ```

**Reply by SMillerDev** [reply]
Why do a whole brew setup? GitHub actions runners already have brew installed. There's also no need to manually specify the url and shasum.

This should be all you need:
```
brew bump-formula-pr screenpipe \
         --version="${VERSION}" \
         --message="Automated update to version ${VERSION}" \
         --no-browse
```

**Reply by krushnarout** [reply]
Got it! Thanks again..
